### PR TITLE
Introduce skolems for virtual term substitution

### PR DESCRIPTION
**Pull request by ajreynol**
This is work towards proof support for virtual term substitution.

### CODE REVIEW

File: include/cvc5/cvc5_skolem_id.h [hunk 1]
```diff
@@ -171,6 +171,40 @@ enum ENUM(SkolemId)
    * - Sort: ``Real``
    */
   EVALUE(TRANSCENDENTAL_SINE_PHASE_SHIFT),
+  /**
+   * Used to reason about virtual term substitution. This term represents
+   * an infinitessemal.
```
Comment (abdoo8080) [inline include/cvc5/cvc5_skolem_id.h:176]:
```suggestion
   * an infinitesimal.
```

File: include/cvc5/cvc5_skolem_id.h [hunk 2]
```diff
@@ -171,6 +171,40 @@ enum ENUM(SkolemId)
...
+   */
+  EVALUE(ARITH_VTS_DELTA),
+  /**
+   * Used to reason about virtual term substitution. This term represents
+   * an infinitessemal.
+   *
+   * - Number of skolem indices: ``0``
+   * - Sort: ``Real``
+   */
+  EVALUE(ARITH_VTS_DELTA_FREE),
```
Comment (abdoo8080) [inline include/cvc5/cvc5_skolem_id.h:189]: How is this different from `ARITH_VTS_DELTA`?
```suggestion
  /**
   * Used to reason about virtual term substitution. This term represents
   * an infinitesimal.
   *
   * - Number of skolem indices: ``0``
   * - Sort: ``Real``
   */
  EVALUE(ARITH_VTS_DELTA_FREE),
```

File: include/cvc5/cvc5_skolem_id.h [hunk 3]
```diff
@@ -171,6 +171,40 @@ enum ENUM(SkolemId)
...
+  EVALUE(ARITH_VTS_INFINITY),
+  /**
+   * Used to reason about virtual term substitution. This term represents
+   * infinity.
+   *
+   * - Number of skolem indices: ``0``
+   *   - ``1:`` A term that represents an arithmetic sort (Int or Real).
+   * - Sort: The sort given by the index.
+   */
+  EVALUE(ARITH_VTS_INFINITY_FREE),
```
Comment (abdoo8080) [inline include/cvc5/cvc5_skolem_id.h:207]: ditto. How is this different from `ARITH_VTS_INFINITY`?

File: src/proof/alf/alf_node_converter.cpp [hunk 4]
```diff
@@ -662,6 +662,7 @@ size_t AlfNodeConverter::getOrAssignIndexForConst(Node v)
 
 bool AlfNodeConverter::isHandledSkolemId(SkolemId id)
 {
+  // Note we don't handle skolems that takes types as arguments yet.
```
Comment (abdoo8080) [inline src/proof/alf/alf_node_converter.cpp:665]:
```suggestion
  // Note we don't handle skolems that take types as arguments yet.
```

**Review by abdoo8080 — APPROVED** [review]

File: include/cvc5/cvc5_skolem_id.h (hunk 3 above)
Comment (ajreynol) [reply include/cvc5/cvc5_skolem_id.h:207]: I will doc.

**Pull request merged**